packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/calendar.tests.js

```diff
@@ -4617,6 +4619,66 @@ QUnit.module('Aria accessibility', {
         fx.off = false;
     }
 }, () => {
+    QUnit.test('aria-label should be \'Readonly-calendar\' and role = group when readOnly option is true', function(assert) {
+        const $element = this.$element;
+
+        const calendar = $element.dxCalendar({
+            readOnly: true
+        }).dxCalendar('instance');
+        const $parentElement = $(calendar._$element);
+        assert.strictEqual($parentElement.attr('aria-label'), 'Readonly-calendar', 'aria-label is set correctly');
+        assert.strictEqual($parentElement.attr('role'), 'group', 'role is set correctly');
+    });
+
+    QUnit.test('aria-label and role should be removed when readOnly option is false', function(assert) {
+        const $element = this.$element;
+
+        const calendar = $element.dxCalendar({
+            readOnly: false
+        }).dxCalendar('instance');
+        const $parentElement = $(calendar._$element);
```

Comment:
Let's not use a private API (_$element)

Reply:
In this case you can use your $element constant, it's the same element. Check this please. So you don't need this variable

Reply:
[212f8b5](https://github.com/DevExpress/DevExtreme/pull/27898/commits/212f8b5f95a5f96df7644e0b287434c920ee75dc)